library/entropy_poll.c

```diff
@@ -86,10 +92,15 @@ int mbedtls_platform_entropy_poll( void *data, unsigned char *output, size_t len
 
 /*
  * Test for Linux getrandom() support.
- * Since there is no wrapper in the libc yet, use the generic syscall wrapper
+ * When glibc > 2.25 include sys/random.h for getrandom(),
+ * else use the generic syscall wrapper
  * available in GNU libc and compatible libc's (eg uClibc).
  */
-#if ((defined(__linux__) && defined(__GLIBC__)) || defined(__midipix__))
+#if HAVE_SYS_RANDOM
+#include <sys/random.h>
+#include <errno.h>
+#define HAVE_GETRANDOM
+#elif defined(__linux__) || defined(__midipix__)
```

Comment:
`syscall` should available on some libc else, not only glibc.
for example musl, uclibc and so on.

Reply:
Indeed, but for example uclibc defines `__GLIBC__` too (when `_GNU_SOURCE` is defined). Also, here we need to be careful about the consequences for libraries that don't support `syscall`: if our condition is too lax, we'll break the build with those libraries. If our condition is too strict, the worst that can happen is we used `/dev/urandom` instead of `syscall`, which is not as bad as failing to build.

So I think this condition should either be:
- `(defined(__linux__) && defined(__GLIBC__)) || defined(__midipix__)` as it was before
- or, if you want to improve on it, `(defined(__linux__) && (defined(__GLIBC__) || defined(__some_other_libc_that_supports_syscal || defined(__yet_another_libc_with_syscall))) || defined(__midipix__)`

But we can't just go ahead and use `syscall`, hoping that perhaps the libc supports it.

Reply:
I pushed with `(defined(__linux__) && defined(__GLIBC__)) || defined(__midipix__)`